src/main/scala/org/apache/spark/rdd/OrderedLeftJoinRDD.scala

```diff
@@ -0,0 +1,44 @@
+package org.apache.spark.rdd
+
+import org.apache.spark._
+import org.broadinstitute.hail.Utils._
+
+import scala.reflect.ClassTag
+
+class OrderedLeftJoinRDD[T, K, V1, V2](rdd1: OrderedRDD[T, K, V1], rdd2: OrderedRDD[T, K, V2])
+  (implicit ev: (K) => T, tOrd: Ordering[T], kOrd: Ordering[K], tct: ClassTag[T],
+    kct: ClassTag[K]) extends RDD[(K, (V1, Option[V2]))](rdd1.sparkContext, Seq(new OneToOneDependency(rdd1),
+  new OrderedDependency(rdd1.orderedPartitioner, rdd2.orderedPartitioner, rdd2)): Seq[Dependency[_]]) {
+
+  override val partitioner: Option[Partitioner] = rdd1.partitioner
+
+  private def p1 = rdd1.orderedPartitioner
+
+  private def p2 = rdd2.orderedPartitioner
+
+  val r2Partitions = rdd2.partitions // compute and store -- getPartitions called on a worker throw a NullPointerExceptione
```

Comment:
I'll move p1 and p2 inside, but r2Partitions has to be computed when the class is constructed.  Calling getPartitions on an executor produces a nullPointerException in ParquetRelation